nexus/server.go

```diff
@@ -488,6 +428,66 @@ type HandlerOptions struct {
 	Serializer Serializer
 }
 
+func (h *httpHandler) handleRequest(writer http.ResponseWriter, request *http.Request) {
+	parts := strings.Split(request.URL.EscapedPath(), "/")
+	// First part is empty (due to leading /)
+	if len(parts) < 3 {
+		h.writeFailure(writer, HandlerErrorf(HandlerErrorTypeNotFound, "not found"))
+		return
+	}
+	service, err := url.PathUnescape(parts[1])
+	if err != nil {
+		h.writeFailure(writer, HandlerErrorf(HandlerErrorTypeBadRequest, "failed to parse URL path"))
+		return
+	}
+	operation, err := url.PathUnescape(parts[2])
+	if err != nil {
+		h.writeFailure(writer, HandlerErrorf(HandlerErrorTypeBadRequest, "failed to parse URL path"))
+		return
+	}
+	var operationID string
+	if len(parts) > 3 {
+		var err error
+		operationID, err = url.PathUnescape(parts[3])
+		if err != nil {
+			h.writeFailure(writer, HandlerErrorf(HandlerErrorTypeBadRequest, "failed to parse URL path"))
+			return
+		}
+	}
+
+	switch len(parts) {
+	case 3: // /{service}/{operation}
+		if request.Method != "POST" {
+			h.writeFailure(writer, HandlerErrorf(HandlerErrorTypeBadRequest, "invalid request method: expected POST, got %q", request.Method))
+			return
+		}
+		h.startOperation(service, operation, writer, request)
+	case 4: // /{service}/{operation}/{operation_id}
+		if request.Method != "GET" {
+			h.writeFailure(writer, HandlerErrorf(HandlerErrorTypeBadRequest, "invalid request method: expected GET, got %q", request.Method))
+			return
+		}
+		h.getOperationInfo(service, operation, operationID, writer, request)
+	case 5:
+		switch parts[4] {
```

Comment:
Can you add a `default` clause here failing?

Reply:
I extracted the fallback outside of the `default:` clause.